usr/bin/irdata_altsplit.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/python3
+
+"""
+Reads the (compressed) <file> in swissprot format,
+and prints the entry if the entry index % <total> == <rank>
+
+--------
+
+This program is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License as published by the Free Software
+Foundation; either version 3 of the License, or (at your option) any later
+version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import gzip
+import re
+
+
+def swiss_prot_entry(textstream, delimiter="^//$"):
```

Comment:
I'd prefer new functions to have docstrings and type annotations. In the case of the `delimiter` argument I would prefer to enforce the invariant that it is a regexp:
```suggestion
def swiss_prot_entry(textstream: Path, delimiter: Pattern[str] = re.compile(r"^//$")) -> Generator[List[str], None, None]:
```

Note in the above a preference for `pathlib.Path`. I'm fine with you ignoring this and typing as `str`.